File: core/src/main/java/io/github/msameer0/rhythmicrush/game/registries/Registry.java
package io.github.msameer0.rhythmicrush.game.registries;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Annotation to mark a class as a registered game element.
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
public @interface Registry {
    /**
     * The unique identifier for this element (e.g., "spike", "ship_portal").
     */
    String id();
}

File: core/src/main/kotlin/io/github/msameer0/rhythmicrush/game/registries/Registry.kt
package io.github.msameer0.rhythmicrush.game.registries

/**
 * Annotation used to define a unique string ID for a class to be used in a GameRegistry.
 */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS)
annotation class Registry(
    val id: String
)
